stories/demos/Grouping.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState, useMemo } from 'react';
+import { groupBy as rowGrouper } from 'lodash';
+import Select, { components, ValueType, OptionsType, Props as SelectProps } from 'react-select';
+import { SortableContainer, SortableElement } from 'react-sortable-hoc';
+import faker from 'faker';
+
+import DataGrid, { Column, Row, SelectColumn } from '../../src';
+import './Grouping.less';
+
+interface Row {
+  id: number;
+  country: string;
+  year: number;
+  sport: string;
+  athlete: string;
+  gold: number;
+  silver: number;
+  bronze: number;
+}
+
+interface Option {
+  value: string;
+  label: string;
+}
+
+const sports = ['Swimming', 'Gymnastics', 'Speed Skating', 'Cross Country Skiing', 'Short-Track Speed Skating', 'Diving', 'Cycling', 'Biathlon', 'Alpine Skiing', 'Ski Jumping', 'Nordic Combined', 'Athletics', 'Table Tennis', 'Tennis', 'Synchronized Swimming', 'Shooting', 'Rowing', 'Fencing', 'Equestrian', 'Canoeing', 'Bobsleigh', 'Badminton', 'Archery', 'Wrestling', 'Weightlifting', 'Waterpolo', 'Wrestling', 'Weightlifting'];
+
+const columns: Column<Row>[] = [
+  SelectColumn,
+  {
+    key: 'country',
+    name: 'Country'
+  },
+  {
+    key: 'year',
+    name: 'Year'
+  },
+  {
+    key: 'sport',
+    name: 'Sport'
+  },
+  {
+    key: 'athlete',
+    name: 'Athlete'
+  },
+  {
+    key: 'gold',
+    name: 'Gold',
+    groupFormatter({ childRows }) {
+      return <>{childRows.reduce((prev, { gold }) => prev + gold, 0)}</>;
+    }
+  },
+  {
+    key: 'silver',
+    name: 'Silver',
+    groupFormatter({ childRows }) {
+      return <>{childRows.reduce((prev, { silver }) => prev + silver, 0)}</>;
+    }
+  },
+  {
+    key: 'bronze',
+    name: 'Bronze',
+    groupFormatter({ childRows }) {
+      return <>{childRows.reduce((prev, { silver }) => prev + silver, 0)}</>;
+    }
+  },
+  {
+    key: 'total',
+    name: 'Total',
+    formatter({ row }) {
+      return <>{row.gold + row.silver + row.bronze}</>;
+    },
+    groupFormatter({ childRows }) {
+      return <>{childRows.reduce((prev, row) => prev + row.gold + row.silver + row.bronze, 0)}</>;
+    }
+  }
+];
+
+function createRows(): Row[] {
+  const rows: Row[] = [];
+  for (let i = 1; i < 10000; i++) {
+    rows.push({
+      id: i,
+      year: 2015 + faker.random.number(3),
+      country: faker.address.country(),
+      sport: sports[faker.random.number(sports.length - 1)],
+      athlete: faker.name.findName(),
+      gold: faker.random.number(5),
+      silver: faker.random.number(5),
+      bronze: faker.random.number(5)
+    });
+  }
+
+  return rows.sort((r1, r2) => r2.country.localeCompare(r1.country));
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#Performance
If you want.